### DISCUSSION TITLE
Consider Making `Ignore Space In Chords` An Option

### DISCUSSION THREAD

**Discussion by Danc2050** [psoukie/zipchord#143] [Ideas]
I find that if I want to quickly type a chord and move onto the next one, when I hit "space" soon after, it will think that a chord has not been typed since it is also registering the space as part of a chord being typed. 

I can point out simple chords that come in ZipChorder like `th` or `wl` or `30`, these are good examples of something I would like to hit at the same time in combination with space (or space soon after) and move onto the next word/letter quickly.

There are probably a few options. Here are some that come to mind
1) make ZipChord faster using DLL calls and/or rewrite in C++ or maybe IronAHK for portability (all of this is a big job)
2) Allow a "ignore space in chords" option or something like that.

I can trigger this issue at 80, 50, and 30ms detection delay.

**Comment by psoukie** [comment]
Thank you! I'll post instructions on how you to check what's happening with the new testing automation. My guess is it is as you write that the spacebar overlaps with the previous keys, so it is understood as part of the chord. (The easiest way to test this is to check the "Delete mistyped chords" option on Detection tab and see if those chords with space get deleted.

A solution would be to implement 2 -- ignore (disallow) space in chords. (Sounds reasonable, and could be done without clutter.) As a workaround, either a longer detection delay should help, or use smart spaces after chords.

**Reply by Danc2050** [reply]
Most likely. Ok, I can see how to do that with the visualizer that was just released probably.

OK neat. I have turned on smart spaces. Thanks for helping with my newby questions!

**Comment by psoukie** [comment]
I've resume work on ZipChord,  and I'm still thinking about this one. From the description, I understand you do not want to use the Add space before / after chords. The new engine in v2.2 is able to differentiate more things, e.g. when the last character of a word and the space overlap too much, in 2.1, the word would not get recognized for hints or expansion of shorthand, but in 2.2 it does. (Unless your settings allow unrestricted chords and deletion of unrecognized.)  The combinations can get tricky, but the new engine has more granularity in differentiating the sequences.
